netket/graph/abstract_graph.py

```diff
@@ -30,8 +35,8 @@ def is_bipartite(self) -> bool:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def edges(self) -> Iterator[Tuple[int, int]]:
-        r"""Iterator over the edges of the graph"""
+    def edges(self, color: Union[bool, int] = False) -> EdgeSequence:
```

Comment:
sorry why the default value is False and not None for the color?

Reply:
`color=True` returns the edge colors, `color=False` doesn't, so it makes sense.

Reply:
but why ```Union[bool,int]``` then?

Reply:
Because `color=0` etc. are valid and filter by color.

(The interface is bit weird with the dual role of `color`. It has been that way for a while though, I think we did it this way to resemble `networkx`?)

Reply:
uhm I see, now I a remember I guess... but indeed it's a bit weird that color is both a filter and a ```return_color``` flag... is there something similar in igraph?

Reply:
The igraph interface provides a dictionary-like interface to edge (and graph/vertex) attributes, so that one can e.g. access edge color as
```python
graph.es["color"]
```

Maybe we should deprecate the color parameter? NetKet v2 did have "edge_colors" as a separate attribute anyways, I think.

Reply:
I think we should make this part of the interface more consistent.
Also I completely dislike that `.edges` is a method and not an attribute.
Maybe we should have two different functions?

Reply:
So lets just have `.edges` return the edges and add again `.edge_colors` for the colors? I'd be happy to do this. (Consistency with `networkx` no longer makes sense for us to pursue anyways.)

Reply:
Yes. I’d be fully in favor with that.

Reply:
Let's do this in a separate PR though